cmd/agent/Dockerfile

```diff
@@ -3,8 +3,11 @@ RUN apk add --update --no-cache ca-certificates
 
 FROM scratch
 ARG TARGETARCH=amd64
+ARG USER_UID=10001
 COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
 
 EXPOSE 5775/udp 6831/udp 6832/udp 5778
 COPY agent-linux-$TARGETARCH /go/bin/agent-linux
 ENTRYPOINT ["/go/bin/agent-linux"]
+
+USER ${USER_UID}
```

Comment:
Just curious, does this UID have a name? How will it show in `ps`?

Reply:
If there's no name/passwd entry associated with the UID, then the process' user should appear as `unknown` in `ps`. If you run kubectl exec, bash will usually show a prompt like `I have no name!@hostname:`. 

The advantage of using a numeric user vs an actual name is that if a Pod Security Policy in Kubernetes requires a container to be run as root (`runAsNonRoot: true`) then setting any "user" other than 0 here allows the container to be run without having to override the user id in the securityContext.

Reply:
For completeness: Kubernetes platforms like OpenShift would typically assign a non-root user dynamically anyway. I expect the same to be true for GKE and AKS.